manifests/params.pp

```diff
@@ -4,9 +4,9 @@
     fail('This module only works on Debian or derivatives like Ubuntu')
   }
 
-  # prior to puppet 3.5.0, defined couldn't test if a variable was defined
-  # strict variables wasn't added until 3.5.0, so this should be fine.
-  if ! $::settings::strict_variables {
+  # prior to puppet 3.5.0, defined() couldn't test if a variable was defined.
+  # strict_variables wasn't added until 3.5.0, so this should be fine.
+  if $::puppetversion and versioncmp($::puppetversion, '3.5.0') < 0 {
```

Comment:
It can be not set in rspec-puppet tests at least :). But no, not in the real world afaik. If not set in spec tests it does the `versioncmp(): a is not a string` or some error message like that.